### ISSUE TITLE
Collector for safety-gym cost key entry?

### ISSUE THREAD

**Issue by drozzy** [thu-ml/tianshou#363]
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [x] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates

Hi, I'm curious how I would implement the tracking of the additional metric `cost` (which is implemented in [safety-gym](https://github.com/openai/safety-gym) environments.

Basically it is contained inside the `info['cost']` objects.
Right now collectors in tianshou discard the `info` dict, and only return a predetermined set of metrics: https://github.com/thu-ml/tianshou/blob/84f58636eb3895413c82cc3a991bf0f0e5b043ad/tianshou/data/collector.py#L159

Would it be possible to extend it to be a bit more general or should I write a new collector class for my needs?

Thank you.

**Comment by Trinkle23897** [comment]
See https://tianshou.readthedocs.io/en/master/tutorials/cheatsheet.html#handle-batched-data-stream-in-collector
